### PR TITLE
fix: developer experience: unify ts/js/tailwind config files between search and chat folders

### LINKED ISSUE #431: developer experience: unify ts/js/tailwind config files between `search` and `chat` folders
Right now there are multiple config files present between `chat` and `search` which are causing issues. We would like to modify the folder such that the configurations are unified. Here are a list of files to remove and place in the root dir: 

- [ ] `postcss.config.cjs`
- [ ] `prettier.cofnig.cjs`
- [ ] `tailwind.config.cjs`
- [ ] `tsconfig.json`

## Bounty acceptance requirement

We must be able to work in the `search` and `chat` projects when we have `arguflow/arguflow` open and not see any red squiggles.

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
fix #431
/claim #431

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,6 +1,12 @@
 {
+  "python.formatting.provider": "none",
```
Comment (skeptrunedev) [inline .vscode/settings.json:2]: I think this needs to be removed: 

image

**Review by skeptrunedev — CHANGES_REQUESTED** [review]
One comment then please squash this into one commit: 

```
git reset --soft @~3
git commit -a -m "fix: developer experience: unify ts/js/tailwind config files between search and chat folders" 
git push --force
```

and star the repo! 

image

**Comment by rishi-raj-jain** [comment]
Starred and done with squashing!

**Comment by skeptrunedev** [comment]
Hey, we don't allow `

> Starred and done with squashing!

Perfect, thank you!

**Comment by rishi-raj-jain** [comment]
> Hey, we don't allow `

What's that?

**Comment by skeptrunedev** [comment]
> > Hey, we don't allow `
> 
> What's that?

So you made merge commits early in the history on this PR and then reset and force pushed which means the diff is now massive when you really only changed a few files. 

We don't allow `merge` commits in our repository, so this is going to have to be fixed. 

At this point, it is likely easiest to just make a new PR. 

I am willing to do that for you as I know it's your first time contributing to our repo and still give you the bounty unless you want to do it?

**Comment by rishi-raj-jain** [comment]
Pushed to the latest update as it was adding some unknown files to commit.

**Comment by rishi-raj-jain** [comment]
@skeptrunedev this should be good now

**Review by skeptrunedev — CHANGES_REQUESTED** [review]
Outstanding issue of the merge commit. Once that is resolved this can be merged.

**Comment by rishi-raj-jain** [comment]
> Outstanding issue of the merge commit. Once that is resolved this can be merged.

Can you re-review again?

**Comment by rishi-raj-jain** [comment]
https://github.com/arguflow/arguflow/pull/449/files

**Comment by skeptrunedev** [comment]
> https://github.com/arguflow/arguflow/pull/449/files

The diff is good now, but we have the merge commit causing an issue: 

image


I am resolving on my end right now.

**Comment by rishi-raj-jain** [comment]
Ah oh okay okay... shall I create a new PR?

**Comment by skeptrunedev** [comment]
> Ah oh okay okay... shall I create a new PR?

Nah, one second.

**Comment by skeptrunedev** [comment]
@rishi-raj-jain all good now, for context here is the way to fix: 

```
git reset --hard <commit hash before the merge>
git rebase main
git push --force
```

**Review by skeptrunedev — APPROVED** [review]
Great work!

**Comment by rishi-raj-jain** [comment]
Screenshot 2023-09-19 at 10 55 46 AM

the twitter icon is not going to anywhere, maybe you want to change it to https://twitter.com/arguflow

Can do a PR, lmk the repo

**Comment by rishi-raj-jain** [comment]
> @rishi-raj-jain all good now, for context here is the way to fix:
> 
> ```
> git reset --hard <commit hash before the merge>
> git rebase main
> git push --force
> ```

Ah I see, I usually merge the latest to test if there's no conflict with what's latest on the repo from the time I started working on it

**Comment by skeptrunedev** [comment]
> Screenshot 2023-09-19 at 10 55 46 AM
> the twitter icon is not going to anywhere, maybe you want to change it to https://twitter.com/arguflow
> 
> Can do a PR, lmk the repo

That would be a blessing! Thanks for pointing it out! - https://github.com/arguflow/landing

**Comment by rishi-raj-jain** [comment]
> Great work!

Loved working on it, thank you!

**Comment by rishi-raj-jain** [comment]
> > Screenshot 2023-09-19 at 10 55 46 AM
> > the twitter icon is not going to anywhere, maybe you want to change it to https://twitter.com/arguflow
> > Can do a PR, lmk the repo
> 
> That would be a blessing! Thanks for pointing it out! - https://github.com/arguflow/landing

https://github.com/arguflow/landing/pull/1 here you go!

**Pull request merged**